### PR TITLE
adding script for comparing API coverage + table at the documentation page

### PR DESCRIPTION
**Pull request by ipdemes**
@magnatelee please review

### CODE REVIEW

**Comment by ipdemes** [comment]
The output table looks like:
Screen Shot 2022-02-16 at 11 15 12 AM

File: docs/cunumeric/source/comparison/_comparison_generator.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    # buf += _section('Polynomial (laguerre)', '.polynomial.laguerre', other_lib)
+    # buf += _section('Masked array operations',".ma", other_lib)
+    # buf += _section('Compatibility module', ".compat", other_lib)
+    # buf += _section('String operations', ".core.defchararray", other_lib)
+    # buf += _section('Numeric Types','.core.numerictypes' , other_lib)
+
+    return "\n".join(buf)
+
+
+# if __name__ == "__main__":
```
Comment (magnatelee) [inline docs/cunumeric/source/comparison/_comparison_generator.py:164]: does this need to be commented out?

File: docs/cunumeric/source/comparison/comparison_table.rst.inc [hunk 2]
```diff
@@ -0,0 +1,618 @@
+NumPy vs cuNumeric APIs
```
Comment (magnatelee) [inline docs/cunumeric/source/comparison/comparison_table.rst.inc:1]: Do we need to add this file to the repository? I thought this was auto-generated.

File: docs/cunumeric/source/comparison/_comparison_generator.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        f"NumPy vs cuNumeric APIs",
+        "---------------------------",
+        "",
+    ]
+    buf += _section("Module-Level", "", other_lib)
+    buf += _section("Multi-Dimensional Array", "", other_lib, klass="ndarray")
+    buf += _section("Linear Algebra", ".linalg", other_lib)
+    buf += _section("Discrete Fourier Transform", ".fft", other_lib)
+    buf += _section("Random Sampling", ".random", other_lib)
+    # buf += _section('Polynomial', '.polynomial.polynomial', other_lib)
```
Comment (magnatelee) [inline docs/cunumeric/source/comparison/_comparison_generator.py:150]: Let's remove these commented-out lines. We can always add them back later.

File: docs/cunumeric/source/comparison/_comparison_generator.py (hunk 1 above)
Comment (ipdemes) [reply docs/cunumeric/source/comparison/_comparison_generator.py:164]: It doesn't, will uncomment it

File: docs/cunumeric/source/comparison/comparison_table.rst.inc (hunk 2 above)
Comment (ipdemes) [reply docs/cunumeric/source/comparison/comparison_table.rst.inc:1]: Yes, It is auto generated. I will remove it from the repo.

File: docs/cunumeric/source/comparison/_comparison_generator.py (hunk 3 above)
Comment (ipdemes) [reply docs/cunumeric/source/comparison/_comparison_generator.py:150]: OK

**Pull request merged**